### PR TITLE
Add OBS_BUILD_DIR to docs

### PR DESCRIPTION
**Pull request by EZ64cool**
### Description

Adds OBS_BUILD_DIR to README.md

### Motivation and Context

This is a really useful CMAKE argument that makes iterative debugging so much easier, before finding this I would manually copy/paste binaries and data from the release folder into a local obs-studio directory.
And it's currently missing from the docs.

### How Has This Been Tested?

None needed

### Types of changes

- Documentation (a change to documentation pages)

### Checklist:

- [X] My code has been run through [clang-format](https://github.com/obsproject/obs-studio/blob/master/.clang-format).
- [X] I have read the [**contributing** document](https://github.com/obsproject/obs-studio/blob/master/CONTRIBUTING.rst).
- [X] My code is not on the master branch.
- [X] The code has been tested.
- [X] All commit messages are properly formatted and commits squashed where appropriate.
- [X] I have included updates to all appropriate documentation.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -120,6 +120,7 @@ Additional build system options are available to developers:
 * `ENABLE_QT`: Adds Qt6 support for custom user interface elements (disabled by default)
 * `CODESIGN_IDENTITY`: Name of the Apple Developer certificate that should be used for code signing
 * `CODESIGN_TEAM`: Apple Developer team ID that should be used for code signing
+* `OBS_BUILD_DIR`: Top level directory of an obs-studio build (Optional - but useful for itterative debugging)
```
Comment (RytoEX) [inline README.md:123]:
```suggestion
* `OBS_BUILD_DIR`: Top level directory of an obs-studio build (Optional - but useful for iterative debugging)
```

**Review by RytoEX — CHANGES_REQUESTED** [review]
Always use present tense in commit message and PR titles (Add instead of Added).

**Comment by PatTheMav** [comment]
The variable only exists for legacy purposes, the correct way is to configure the project with an `INSTALL_PREFIX` that points to a directory that a local OBS installation uses to load plugins and just run `cmake --install` to update the installed version.

It was (and is) also only available on Windows, whereas using the `INSTALL_PREFIX` is supported by all platforms. Thus it's not documented, because it's not supposed to be used in the first place.

**Comment by EZ64cool** [comment]
Okay yeah using INSTALL_PREFIX is a good alternative, as a nice bonus I can now do everything from inside VSCode.
Thank you for the tip

**Pull request closed without merging**